programs/fileio.c

```diff
@@ -1651,6 +1645,25 @@ static unsigned long long FIO_getLargestFileSize(const char** inFileNames, unsig
     return maxFileSize;
 }
 
+static const char*
+FIO_determineMirroredDestName(const char* srcFileName, const char* outDirName, const char* suffix)
+{
+    if (!UTIL_isFileNameValidForMirroredOutput(srcFileName))
+        return NULL;
+
+    char* srcFileNameCopy = strdup(srcFileName);
```

Comment:
Be aware that `zstd` uses `C90` convention, for increased portability.
This requires all variables to be declared at the beginning of a block.

Sometimes, this requirement mandates creating new blocks, just for the purpose of declaring the variable,
which is actually useful whenever the variable also has a limited lifespan, 
as the block boundaries give it a clear end of life,
an often overlooked concept in managing software complexity.

Reply:
This project's coding style has a strong preference for `const` property,
which must be consistently applied whenever possible.

In this case, `srcFileNameCopy` should rather be a `char* const` pointer,
meaning that after its initialization, the pointer value never changes.

Reply:
`strdup` is actually creating an object in memory, and transferring its ownership into its return value.
You correctly identified that the memory object must be freed afterwards, which is great !

However, it also means that `strdup` may fail, and return a `NULL` pointer as return value.
This should be checked.

Given that it's not a situation we necessarily want to recover from (if the system can't even allocate enough memory for a simple filename, it's probably in a bad state already), you are allowed to end execution when it happens, in order to reduce the number of scenarios that the rest of the code must deal with.

Reply:
yeah, the C90 "declare before use" caught me be by surprise, I used to 100% following this rule. But after some years of C++, I forget it completely :D